contracts/MultiRoundCheckout.sol

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.17;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+
+contract MultiRoundCheckout is OwnableUpgradeable, PausableUpgradeable, ReentrancyGuardUpgradeable {
```

Comment:
we need to initialize the owner with `__Ownable_init` otherwise it's the `0x0` address.

the best way is to add an initialize function using the openzeppelin `Initializable` contract

Reply:
same for pausable, we need to call `__Pausable_init`

Reply:
we also need to expose an `onlyOwner` function `pause` that calls the internal `_unpause` function
we also need `unpause`, but we can have it has a parameter of the same function

Reply:
in the same init function we need to call `__ReentrancyGuard_init` as well